lib/ibm/cloud/sdk/vpc/http_vpc/vpc_collection.rb

```diff
@@ -0,0 +1,35 @@
+# typed: true
+# frozen_string_literal: true
+
+require_relative '../http_vpc'
+require_relative 'vpc_http'
+
+module IBM
+  module Cloud
+    module SDK
+      module VPC
+        # Container that encapsulates the VPC API.
+        # This class is used as a base for collection APIs.
+        # @param parent [Object] The parent instance in the API chain.
+        # @param endpoint [string] A path from the parent to the desired endpoint. In most cases is should be 1 word.
+        # @param array_key [string] The key that the API response holds the endpoint data. When nil the endpoint will be used.
+        # @param child_class [Object] The Object to be used when instanciating the single instance for this class.
+        class BaseCollection < SDKHTTP::BaseCollection
+          include VpcHTTP
+
+          # Determine if the collection has a total_count key in its response.
+          # @return [Boolean]
+          def has_count?
+            fetch.json&.key?(:total_count)
+          end
+
+          # Get the total count if it exists in the response. Returns nil otherwise.
+          # @return [Integer] The total count reuturned by the server.
+          def count
+            fetch.json&.fetch(:total_count, nil)
```

Comment:
`ary.fetch(x, nil)` can just be `ary[x]`

Reply:
With safe nav operator:

`ary&.fetch(x, nil)` can just be `ary&.[](x)`

Reply:
I put that in because I was getting key errors when the key did not exist. The second parameter is the default parameter. So if the API does not return with t a total_count then nil is returned. Which is testable.

Reply:
I'm not against the sematics, I'm suggesting alternative (preferable?) syntax

Reply:
I'm saying that `[]` is also a standard method and is the less complicated method.  fetch is generally not used in Ruby except when you want it to raise an exception on a missing key...nearly everyone uses `[]`

Reply:
If I understand correctly [](key) and fetch(key, nil) both return the same result. I don't have a strong opinion on either. I can change it later. Can we get this merged?

Reply:
Updated